e2e/pages/events/addHearingBookingDetailsEvent.page.js

```diff
@@ -0,0 +1,58 @@
+const I = actor();
+
+module.exports = {
+  fields: function (index) {
+    return {
+      hearingDetails: {
+        hearingType: {
+          caseManagement: `#hearingDetails_${index}_hearingType-CASE_MANAGEMENT`,
+        },
+        venue: `#hearingDetails_${index}_venue`,
+        date: {
+          day: `#hearingDetails_${index}_hearingDate-day`,
+          month: `#hearingDetails_${index}_hearingDate-month`,
+          year: `#hearingDetails_${index}_hearingDate-year`,
+        },
+        preHearingAttendance: `#hearingDetails_${index}_preHearingAttendance`,
+        hearingTime: `#hearingDetails_${index}_hearingTime`,
+
+        hearingNeedsBooked: {
+          interpreter: `#hearingDetails_${index}_hearingNeededDetails-INTERPRETER`,
+          welsh: `#hearingDetails_${index}_hearingNeededDetails-SPOKEN_OR_WRITTEN_WELSH`,
+          somethingElse: `#hearingDetails_${index}_hearingNeededDetails-SOMETHING_ELSE`,
+        },
+        giveDetails: `#hearingDetails_${index}_hearingNeededGiveDetails`,
+        judgeTitle: `#hearingDetails_${index}_judgeTitle`,
+        judgeFullName: `#hearingDetails_${index}_judgeFullName`,
+      },
+    };
+  },
+
+  addHearingButton: '//*[@id="hearingDetails"]/div/button[1]',
+
+  addHearing() {
+    I.click(this.addHearingButton);
+  },
+
+  async enterHearingDetails(hearingDetails) {
+    const elementIndex = await this.getActiveElementIndex();
+
+    I.click(this.fields(elementIndex).hearingDetails.hearingType);
+    await I.fillField(this.fields(elementIndex).hearingDetails.venue, hearingDetails.venue);
```

Comment:
Do we need to await on line 41, 42?

Reply:
👍

Reply:
Need to have it on the venue as sometimes when clicking the field before it misses venue